From b1fb89055d4d1bc88b7b0b788ad8eca4f3641f9a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Peter=20K=C3=BCmmel?= <syntheticpp@gmx.net>
Date: Fri, 7 Feb 2014 13:12:37 +0100
Subject: [PATCH] Fix linuxfb argument 'mmsize' parsing

Parse first for 'mmsize' because the regex for 'size' also fits to 'mmsize'.

Task-number: QTBUG-29133
Change-Id: Idc4950270818e496d5d94a97a172b7c780f069b1
---
 src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp b/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp
index 735a43d..33a9523 100644
--- a/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp
+++ b/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp
@@ -333,6 +333,8 @@ bool QLinuxFbScreen::initialize()
     foreach (const QString &arg, mArgs) {
         if (arg == QLatin1String("nographicsmodeswitch"))
             doSwitchToGraphicsMode = false;
+        else if (mmSizeRx.indexIn(arg) != -1)
+            userMmSize = QSize(mmSizeRx.cap(1).toInt(), mmSizeRx.cap(2).toInt());
         else if (sizeRx.indexIn(arg) != -1)
             userGeometry.setSize(QSize(sizeRx.cap(1).toInt(), sizeRx.cap(2).toInt()));
         else if (offsetRx.indexIn(arg) != -1)
@@ -341,8 +343,6 @@ bool QLinuxFbScreen::initialize()
             ttyDevice = ttyRx.cap(1);
         else if (fbRx.indexIn(arg) != -1)
             fbDevice = fbRx.cap(1);
-        else if (mmSizeRx.indexIn(arg) != -1)
-            userMmSize = QSize(mmSizeRx.cap(1).toInt(), mmSizeRx.cap(2).toInt());
     }
 
     if (fbDevice.isEmpty()) {
-- 
1.8.3.2

